snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -513,14 +525,29 @@ def _generate_desktop_file(self, name, app):
             desktop_file.parse_and_reformat()
             desktop_file.write(gui_dir=os.path.join(self.meta_dir, 'gui'))
 
-    def _render_socket_modes(self, apps):
+    def _render_socket_modes(self, apps: Dict[str, Any]) -> None:
         for app in apps.values():
             sockets = app.get('sockets', {})
             for socket in sockets.values():
                 mode = socket.get('socket-mode')
                 if mode is not None:
                     socket['socket-mode'] = OctInt(mode)
-        return apps
+
+    def _apply_pass_through(self, location: str, this: Dict[str, Any],
+                            pass_through: Dict[str, Any],
+                            original: Dict[str, Any]) -> None:
+        # Look for properties defined in pass-through and add them to the
+        # given dictionary. Any value already in the original dictionary must
+        # not be specified in pass-through at the same time.
+        for key in pass_through:
+            if key in original:
+                raise meta_errors.AmbiguousPassThroughKeyError(key)
```

Comment:
If one has multiple keys overlapping, this error will result in a bit of a whack-a-mole experience. May I suggest instead turning each collection of keys into a set and intersecting them, and raising an error with _all_ conflicting keys?